agent/lib/kontena/fluentd/server_connection.rb

```diff
@@ -0,0 +1,34 @@
+require_relative '../helpers/rpc_helper'
+
+module Kontena
+  module Fluentd
+    class ServerConnection
+      include Celluloid
+      include Kontena::Logging
+
+      finalizer :close
+
+      def initialize(connection, queue)
+        @peeraddr = connection.peeraddr
+        info "connection opened from #{@peeraddr[1]}:#{@peeraddr[2]}"
+        @connection = connection
+        @queue = queue
+        async.receive
+      end
+
+      def receive
+        unpacker = MessagePack::Unpacker.new(@connection)
```

Comment:
Needs to be in a separate `defer { ... }` thread or this will block the actor, preventing the server from terminating it?